sdk/tests/utests/Job_tests.cpp

```diff
@@ -0,0 +1,279 @@
+        scavengerCV.wait(lock, [&] { return startedWaiting; });
+        std::this_thread::sleep_for(2ms);
+    }
+
+    EXPECT_FALSE(finishedWaiting);
+
+    // Notify executor to finish
+    {
+        std::lock_guard guard{executorMutex};
+        finishExecution = true;
+        executorCV.notify_one();
+    }
+    executor.join();
+    scavenger.join();
+}
+
+TEST(Test_Job, finishExecution_waitingForTermination_returnsImmediatly) {
+    bool                    isExecuting     = false;
+    bool                    finishExecution = false;
+    std::mutex              executorMutex;
+    std::condition_variable executorCV;
+    Job                     job([&] {
+        {
+            std::lock_guard guard{executorMutex};
+            isExecuting = true;
+            executorCV.notify_one();
+        }
+        {
+            std::unique_lock lock(executorMutex);
+            executorCV.wait(lock, [&] { return finishExecution; });
+        }
+    });
+    std::thread             executor([&] {
+        job.execute();
+        {
+            std::lock_guard guard{executorMutex};
+            isExecuting = false;
+            executorCV.notify_one();
+        }
+    });
+
+    // make sure job is in execution
+    {
+        std::unique_lock lock(executorMutex);
+        executorCV.wait(lock, [&] { return isExecuting; });
+    }
+
+    // setup the scavenger (waiting for termination of the job)
+    bool                    startedWaiting  = false;
+    bool                    finishedWaiting = false;
+    std::mutex              scavengerMutex;
+    std::condition_variable scavengerCV;
+    std::thread             scavenger([&] {
+        {
+            std::lock_guard guard{scavengerMutex};
+            startedWaiting = true;
+        }
+        scavengerCV.notify_one();
+        job.waitForTermination();
+        {
+            std::lock_guard guard{scavengerMutex};
+            finishedWaiting = true;
+        }
+        scavengerCV.notify_one();
+    });
+
+    // make sure scavenger is waiting
+    {
+        std::unique_lock lock{scavengerMutex};
+        scavengerCV.wait(lock, [&] { return startedWaiting; });
+        std::this_thread::sleep_for(2ms);
+    }
+
+    // Notify executor to finish
+    {
+        std::lock_guard guard{executorMutex};
+        finishExecution = true;
+        executorCV.notify_one();
+    }
+
+    // Wait till executor is done
+    {
+        std::unique_lock lock{executorMutex};
+        executorCV.wait(lock, [&] { return !isExecuting; });
+    }
+
+    // make sure scavenger is done
+    {
+        std::unique_lock lock{scavengerMutex};
+        scavengerCV.wait(lock, [&] { return finishedWaiting; });
+        EXPECT_TRUE(finishedWaiting);
+    }
+
+    executor.join();
+    scavenger.join();
+}
+
+TEST(Test_RecurringJob, callCtor_none_noFailure) {
+    EXPECT_NO_THROW(RecurringJob([] {}));
+}
+
+TEST(Test_RecurringJob, create_firstCall_noFailureAndNoNullPtr) {
```

Comment:
Stay positive with the expectations

```suggestion
TEST(Test_RecurringJob, create_firstCall_validPtr) {
```

Reply:
That's about my pedantic nature: How to prove that the pointer "is valid"?